src/js/core/methods/helpers/signtx.js

```diff
@@ -5,202 +5,193 @@ import type { RefTransaction, TransactionOptions, SignedTransaction } from '../.
 
 import type {
     TypedCall,
-    TxAckResponse,
     TxRequest,
-    RequestType,
     TxInputType,
     TxOutputType,
     TxRequestSerializedType,
 } from '../../../types/trezor/protobuf';
 
-const requestPrevTxInfo = (
-    reqTx: RefTransaction,
-    requestType: RequestType,
-    requestIndex: string | number,
-    dataLen: ?(string | number),
-    dataOffset: ?(string | number),
-): TxAckResponse => {
-    const i = +requestIndex;
-    if (requestType === 'TXINPUT') {
-        return {inputs: [reqTx.inputs[i]]};
-    }
-    if (requestType === 'TXOUTPUT') {
-        return {bin_outputs: [reqTx.bin_outputs[i]]};
-    }
-    if (requestType === 'TXEXTRADATA') {
-        if (dataLen == null) {
+type RefTxs = { [hash: string]: RefTransaction };
+type Props = {
+    typedCall: TypedCall;
+    txRequest: TxRequest;
+    refTxs: RefTxs;
+    inputs: TxInputType[];
+    outputs: TxOutputType[];
+    serializedTx: string[];
+    signatures: string[];
+};
+
+const requestPrevTxInfo = ({
+    typedCall,
+    txRequest: { request_type, details },
+    refTxs,
+}: Props) => {
+    const { tx_hash } = details;
+    if (!tx_hash) {
+        throw ERRORS.TypedError('Runtime', 'requestPrevTxInfo: unknown details.tx_hash');
+    }
+    const tx = refTxs[tx_hash.toLowerCase()];
+    if (!tx) {
+        throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown tx: ${tx_hash}`);
+    }
+    if (request_type === 'TXINPUT') {
+        if (!tx.bin_outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXINPUT: ${tx_hash}`);
+        return typedCall('TxAckPrevInput', 'TxRequest', { tx: { input: tx.inputs[details.request_index] } });
+    }
+    if (request_type === 'TXOUTPUT') {
+        if (!tx.bin_outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXOUTPUT: ${tx_hash}`);
+        return typedCall('TxAckPrevOutput', 'TxRequest', { tx: { output: tx.bin_outputs[details.request_index] } });
+    }
+    if (request_type === 'TXORIGINPUT') {
+        if (!tx.outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXORIGINPUT: ${tx_hash}`);
+        return typedCall('TxAckInput', 'TxRequest', { tx: { input: tx.inputs[details.request_index] } });
+    }
+    if (request_type === 'TXORIGOUTPUT' && tx.outputs) {
+        if (!tx.outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXORIGOUTPUT: ${tx_hash}`);
```

Comment:
this is weird, you already have && tx.outputs on line 50. is that expected?

Reply:
good catch, it shouldn't be in line 50

Reply:
addressed 9e1c7a376ff627be22f3fd2d690faeec7eccfffb